src/java.base/share/classes/sun/security/ssl/SSLCipher.java

```diff
@@ -1859,10 +1859,23 @@ public Plaintext decrypt(byte contentType, ByteBuffer bb,
                 }
 
                 if (bb.remaining() <= tagSize) {
-                    throw new BadPaddingException(
-                        "Insufficient buffer remaining for AEAD cipher " +
-                        "fragment (" + bb.remaining() + "). Needs to be " +
-                        "more than tag size (" + tagSize + ")");
+                    // Check for unexpected plaintext alert.
+                    if (ContentType.ALERT.equals(ContentType.valueOf(contentType))) {
+                        // In TLSv1.3 alert level can be ignored, we only get the alert.
+                        final String msg = "Unexpected plaintext alert received: "
+                                + Alert.nameOf(bb.get(bb.position() + 1));
+
+                        if (SSLLogger.isOn && SSLLogger.isOn("ssl")) {
+                            SSLLogger.info(msg);
+                        }
```

Comment:
It may not need the log any longer because the follow-up exception will cover the information.

Reply:
It may be sufficient to have it in exception only.  We normally don't log exception message alone in SunJSSE.  From my understand, it is hardly to see a lot benefits.

Reply:
I don't think we know the record is "plaintext", even it has just two bytes.  The input is not verified, it can be anything.

Reply:
We assume that it's a plain text because `ContentType` is not of `APPLICATION_DATA` type and data is less than `tagSize`. This is the same assumption that is made in RFC:

`The server then ignores early data by skipping all records with an external
 content type of "**application_data**" (indicating that they are **encrypted**)`

In any case, we will provide the integer values of those 2 bytes if they are not known alert level and alert type.